packages/siri-sx-importer/dynamo.ts

```diff
@@ -0,0 +1,82 @@
+import { Consequence, DisruptionInfo } from "@create-disruptions-data/shared-ts/disruptionTypes";
+import * as console from "console";
+import { Organisation, organisationSchema } from "./utils/importerSiriTypes.zod";
+
+const ddbDocClient = DynamoDBDocumentClient.from(new DynamoDBClient({ region: "eu-west-2" }));
+
+export const getOrgIdFromDynamo = async (participantRef: string, tableName: string): Promise<string> => {
+    const dbData = await ddbDocClient.send(
+        new ScanCommand({
+            TableName: tableName,
+        }),
+    );
+
+    const parsedOrg = organisationSchema.parse(dbData.Items);
+
+    const filteredOrg: Organisation = parsedOrg.filter((item) => item.name === participantRef);
+
+    return !!filteredOrg ? filteredOrg[0].PK : "";
+};
+
+const createConsequenceBatches = (items: Consequence[], size = 50) =>
+    Array.from({ length: Math.ceil(items.length / size) }, (v, i) => items.slice(i * size, i * size + size));
+
+const createDisruptionInfoBatches = (items: DisruptionInfo[], size = 50) =>
+    Array.from({ length: Math.ceil(items.length / size) }, (v, i) => items.slice(i * size, i * size + size));
+
+export const publishDisruptionAndConsequenceInfoToDynamo = (disruptionInfo: DisruptionInfo[], consequenceInfo: Consequence[], tableName: string) => {
+    const disruptionBatches = disruptionInfo.length > 50 ? createDisruptionInfoBatches(disruptionInfo) : [disruptionInfo];
+    const consequenceBatches = consequenceInfo.length > 50 ? createConsequenceBatches(consequenceInfo) : [consequenceInfo];
+
+    disruptionBatches.forEach((batch) => {
+        const disruptionsPutCommand = batch.map((disruption) => {
+            return {
+                Put: {
+                    TableName: tableName,
+                    Item: {
+                        PK: disruption.orgId,
+                        SK: `${disruption.disruptionId}#INFO`,
+                        publishStatus: "PUBLISHED",
+                        ...disruption,
+                    },
+                },
+            };
+        })
+
+        ddbDocClient
+            .send(
+                new TransactWriteCommand({
+                    TransactItems: [...disruptionsPutCommand],
+                }),
+            )
+            // eslint-disable-next-line no-console
+            .catch((e) => console.error(e));
+    })
+
+    consequenceBatches.forEach((batch) => {
+        const consequencePutCommand = batch.map((consequence) => {
+            return {
+                Put: {
+                    TableName: tableName,
+                    Item: {
+                        PK: consequence.orgId,
+                        SK: `${consequence.disruptionId}#CONSEQUENCE#${consequence.consequenceIndex}`,
+                        publishStatus: "PUBLISHED",
+                        ...consequence,
+                    },
+                },
+            };
+        });
+
+        ddbDocClient
+            .send(
+                new TransactWriteCommand({
```

Comment:
This will still be doing 2 separate transactions, we'd need something like:
```ts
const isConsequenceInfo = (item: Consequence | DisruptionInfo): item is Consequence =>
    !!(item as Consequence).consequenceIndex;

const createBatches = (items: (Consequence | DisruptionInfo)[], size = 50) =>
    Array.from({ length: Math.ceil(items.length / size) }, (v, i) => items.slice(i * size, i * size + size));

export const publishDisruptionAndConsequenceInfoToDynamo = (
    disruptionInfo: DisruptionInfo[],
    consequenceInfo: Consequence[],
    tableName: string,
) => {
    const mergedItems = [...disruptionInfo, ...consequenceInfo];
    const batches = mergedItems.length > 50 ? createBatches(mergedItems) : [mergedItems];

    batches.forEach((batch) => {
        const disruptionsPutCommand = batch.map((item) => {
            if (isConsequenceInfo(item)) {
                return {
                    Put: {
                        TableName: tableName,
                        Item: {
                            PK: item.orgId,
                            SK: `${item.disruptionId}#CONSEQUENCE#${item.consequenceIndex}`,
                            publishStatus: "PUBLISHED",
                            ...item,
                        },
                    },
                };
            }

            return {
                Put: {
                    TableName: tableName,
                    Item: {
                        PK: item.orgId,
                        SK: `${item.disruptionId}#INFO`,
                        publishStatus: "PUBLISHED",
                        ...item,
                    },
                },
            };
        });

        ddbDocClient
            .send(
                new TransactWriteCommand({
                    TransactItems: [...disruptionsPutCommand],
                }),
            )
            // eslint-disable-next-line no-console
            .catch((e) => console.error(e));
    });
};
```